src/shogun/kernel/string/HistogramWordStringKernel.cpp

```diff
@@ -434,14 +434,8 @@ void HistogramWordStringKernel::init()
 	/*m_parameters->add_vector(&variance, &num_params2, "variance");*/
 	watch_param("variance", &variance, &num_params2);
 
-	SG_ADD(&estimate, "estimate", "Plugin Estimate.");
-
-	add_callback_function("estimate", [this](){
-		if (!std::dynamic_pointer_cast<PluginEstimate>(estimate)) {
-			error("Expected Machine to be PluginEstimate");
-			estimate.reset();
-		}
-	});
+	SG_ADD(&estimate, "estimate", "Plugin Estimate.", ParameterProperties::CONSTRAIN,
+			SG_CONSTRAINT(dynamic_cast_checker<std::shared_ptr<Machine>, PluginEstimate>()));
```

Comment:
@jonpsy so here you go, you pass the type you are registering in the first template argument and then the type it must be castable to at runtime :)

Reply:
Thank you so much!!

Reply:
and then you are guaranteed to have defined behaviour if the `put` fails